src/actions/sso/index.ts

```diff
@@ -29,5 +50,59 @@ export const consumeCredentialRequest = (jwtEncodedCR: string) => {
         credentials: attributeSummaries
       }
     }))
+
+    const summary = {
+      requester: CR.getRequester(),
+      callbackURL: CR.getCallbackURL(),
+      request: credentialRequests
+    }
+
+    dispatch(setCredentialRequest(summary))
+    dispatch(navigationActions.navigate({routeName: routeList.Consent}))
+  }
+}
+
+// TODO Decrypt when fetching from storage
+export const sendCredentialResponse = (selectedCredentials: StateVerificationSummary[]) => {
+  return async(dispatch: Dispatch<AnyAction>, getState: Function, backendMiddleware: BackendMiddleware) => {
+    const { jolocomLib, storageLib, keyChainLib, encryptionLib, ethereumLib } = backendMiddleware
+
+    const encryptionPass = await keyChainLib.getPassword()
+    const currentDid = getState().account.did.get('did')
+    const personaData = await storageLib.get.persona({did: currentDid})
+    const { encryptedWif } = personaData[0].controllingKey
+
+    const decryptedWif = encryptionLib.decryptWithPass({
+      cipher: encryptedWif,
+      pass: encryptionPass
+    })
+
+    const { privateKey } = ethereumLib.wifToEthereumKey(decryptedWif)
+
+    const wallet = jolocomLib.wallet.fromPrivateKey(Buffer.from(privateKey, 'hex'))
+
+    const credentials = await Promise.all(selectedCredentials.map(async cred => {
+      const results = await storageLib.get.verifiableCredential({id: cred.id})
+      return results[0]
+    }))
+
+    const jsonForm = credentials.map(cred => cred.toJSON())
+    const credentialResponse = wallet.createCredentialResponse(jsonForm)
+
+    const { callbackURL } = getState().sso.activeCredentialRequest
+
+    // TODO Do we care about the response?
+    try {
+      await fetch(callbackURL, {
+        method: 'POST',
+        body: JSON.stringify({token: credentialResponse}),
+        headers: {'content-type': 'application/json'}
+      })
+
+      dispatch(clearCredentialRequest())
+      dispatch(navigationActions.navigate({ routeName: routeList.Identity }))
```

Comment:
maybe there should be a navigation reset action dispatched here as well, so the consent screen is no longer on the navigation stack.